pkg/master/controller.go

```diff
@@ -117,6 +117,9 @@ func (c *Controller) UpdateKubernetesService() error {
 		if err := c.CreateMasterServiceIfNeeded("kubernetes", c.ServiceIP, servicePorts, serviceType); err != nil {
 			return err
 		}
+		if err := c.SetService("kubernetes", servicePorts, serviceType); err != nil {
```

Comment:
I can move the `SetService` call into what would be `CreateOrUpdateMasterServiceIfNeeded` but I'd like to stick with the `SetService` function as a compartmental thing to test.